src/main/java/org/testng/TestRunner.java

```diff
@@ -728,18 +738,29 @@ private void privateRun(XmlTest xmlTest) {
         }
       } else {
         List<ITestNGMethod> freeNodes = graph.getFreeNodes();
+        if (true) {
```

Comment:
Just seeing these `if(true)` blocks here, I'll remove those when I get a chance.

Reply:
There's no way to completely remove the sorting. The sort right below the line you commented on makes sure that cases with multiple non-dependent methods of different priorities (like https://github.com/cbeust/testng/blob/master/src/test/java/test/dependent/github1380/GitHub1380Sample2.java) get run in the correct order (see https://github.com/cbeust/testng/blob/master/src/test/java/test/dependent/DependentTest.java#L156).

Reply:
Well, that's not quite true. If we know that not a single method in the run uses priority, then we don't need to do any sorting.